uproot/_connect/to_pandas.py

```diff
@@ -104,95 +105,82 @@ def futures2df(futures, outputtype, entrystart, entrystop, flatten, flatname):
 
             if isinstance(interpretation, asobj) and isinstance(interpretation.content, astable):
                 interpretation = interpretation.content
             if isinstance(interpretation, astable) and isinstance(interpretation.content, asdtype):
                 interpretation = interpretation.content
 
             if isinstance(interpretation, asjagged):
-                entries = numpy.empty(len(array.content), dtype=numpy.int64)
-                subentries = numpy.empty(len(array.content), dtype=numpy.int64)
-                starts, stops = array.starts, array.stops
-                i = 0
-                numentries = entrystop - entrystart
-                while i < numentries:
-                    entries[starts[i]:stops[i]] = i + entrystart
-                    subentries[starts[i]:stops[i]] = numpy.arange(stops[i] - starts[i])
-                    i += 1
-
-                df = outputtype(index=pandas.MultiIndex.from_arrays([entries, subentries], names=["entry", "subentry"]))
 
                 interpretation = interpretation.content
                 if isinstance(interpretation, asobj) and isinstance(interpretation.content, astable):
                     interpretation = interpretation.content
                 if isinstance(interpretation, astable) and isinstance(interpretation.content, asdtype):
                     interpretation = interpretation.content
 
-                if isinstance(interpretation, asdtype):
-                    if interpretation.todims == ():
-                        if interpretation.todtype.names is None:
-                            fn = flatname(name, None, ())
-                            df[fn] = array.flatten()
-                        else:
-                            for nn in interpretation.todtype.names:
-                                if not nn.startswith(" "):
-                                    fn = flatname(name, nn, ())
-                                    df[fn] = array[nn].flatten()
-                    else:
-                        for tup in itertools.product(*[range(x) for x in interpretation.todims]):
-                            if interpretation.todtype.names is None:
-                                fn = flatname(name, None, tup)
-                                df[fn] = array[(slice(None),) + tup].flatten()
-                            else:
-                                for nn in interpretation.todtype.names:
-                                    if not nn.startswith(" "):
-                                        fn = flatname(name, nn, tup)
-                                        df[fn] = array[nn][(slice(None),) + tup].flatten()
+                array = future()
 
-                else:
-                    fn = flatname(name, None, ())
-                    df[fn] = list(array.flatten())
+                if starts is None:
+                    starts = array.starts
+                    stops = array.stops
+                array = array.flatten()
```

Comment:
I guess my original code did it, but if we're taking `starts` and `stops` as-is, without changing them to `counts` and such, then we're already assuming that `array.content == array.flatten()`. This was naive in uproot-methods, where the arrays could have come from anywhere, but it's probably justified coming directly out of a ROOT file, where `array.content == array.flatten()` may be guaranteed. Since we're making the `starts`, `stops` assumption, we should probably be consistent and use `array.content` as well, just to make that assumption more explicit.

Reply:
I turned `.flatten()` → `.content` and `stops.max()` → `stops[-1]`, strongly making the assumption that `starts` and `stops` are an offsets pair, since they just came from the ROOT file with no user intervention.